### ISSUE TITLE
aliasing when connecting viewport texture to polygon material

### ISSUE THREAD

**Issue by 99alfie** [godotengine/godot#27432]
**Godot version:**
v3.2 custom a53645e

**OS/device including version:**
Windows 10, AMD Radeon Pro WX 7100

**Issue description:**
directly plugging the viewport texture with 2d content into a polygon plane material's albedo in 3d aliases.

**Steps to reproduce:**
using the 2d_in_3d example, it can be seen on the "ball", I added a large font text to make it more obvious in the attached example. I believe the issue is because there is no filtered texture lookup as with a image resource, since the viewport texture just comes in as it is. One way of dealing with it may be writing a shader that does a filtered lookup on the input texture and plugs the result into the output texture. Or is there a easier way I have overlooked?

**Minimal reproduction project:**
[2d_in_3d.zip](https://github.com/godotengine/godot/files/3007215/2d_in_3d.zip)

**Comment by CptPotato** [comment]
You should be able to enable `filter`/`mipmaps` in the properties of the `ViewportTexture`.

**Comment by 99alfie** [comment]
yep you are right, I overlooked the flags param of Texture. Setting it to FLAG_FILTER or FLAGS_DEFAULT (which contains the other) makes things smooth. For the 2d_in_3d example, this does it:

```gdscript
var texture = viewport.get_texture()
texture.flags = Texture.FLAGS_DEFAULT
get_node("Viewport_quad").material_override.albedo_texture = texture
```

**Comment by 99alfie** [comment]
It might be useful though to expose this in the ui for the case where the viewport texture is connected to a polygon's shader through the ui.

**Comment by Calinou** [comment]
Is this covered by the settings available when clicking on a ViewportTexture?

![image](https://user-images.githubusercontent.com/180032/63630130-e597f780-c617-11e9-8214-6182ebe320d2.png)

**Comment by 99alfie** [comment]
not in this case. There is no ViewportRect and TextureRect. Only a Viewport with a Node2D. There is no "Flags" setting on the Viewport node itself, and the texture that Viewport::get_texture() returns comes in with no Filter applied. No way to apply the filter in the UI currently. For me this is fine and might be expected behaviour. However people doing the 2D in 3D example from the docs might wonder why the 2D scene is jaggy-edged and have no UI-way to fix this.

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/17220 (solved in 4.0).
